CMakeLists.txt

```diff
@@ -1,59 +1,43 @@
-#                        if BII_LIB_SRC is empty
-#     ${BII_BLOCK_TARGETS} List of all targets defined in this block
-#     ${BII_BLOCK_EXES} List of executables targets defined in this block
-#     ${BII_exe_name_TARGET}: Executable target (e.g. ${BII_main_TARGET}. You can also use
-#                            directly the name of the executable target (e.g. user_block_main)
-#
-# > EXAMPLE: Add include directories to all targets of this block
-#
-#    TARGET_INCLUDE_DIRECTORIES(${BII_BLOCK_TARGET} INTERFACE myincludedir)
-#
-# > EXAMPLE: Link with pthread:
-#
-#    TARGET_LINK_LIBRARIES(${BII_BLOCK_TARGET} INTERFACE pthread)
-#        or link against library:
-#    TARGET_LINK_LIBRARIES(${BII_LIB_TARGET} PUBLIC pthread)
-#
-#    NOTE:  This can be also done adding pthread to ${BII_LIB_DEPS}
-#            BEFORE calling ADD_BIICODE_TARGETS()
-#
-# > EXAMPLE: how to activate C++11
-#
-#    IF(APPLE)
-#         TARGET_COMPILE_OPTIONS(${BII_BLOCK_TARGET} INTERFACE "-std=c++11 -stdlib=libc++")
-#    ELSEIF (WIN32 OR UNIX)
-#         TARGET_COMPILE_OPTIONS(${BII_BLOCK_TARGET} INTERFACE "-std=c++11")
-#    ENDIF(APPLE)
-#
-# > EXAMPLE: Set properties to target
-#
-#    SET_TARGET_PROPERTIES(${BII_BLOCK_TARGET} PROPERTIES COMPILE_DEFINITIONS "IOV_MAX=255")
-#
+
+cmake_minimum_required(VERSION 3.0)
+
+project(FakeIt VERSION 2.0.5 LANGUAGES CXX)
+
+#Create library with INTERFACE parameter because we want to use headers from outside.
+add_library(FakeIt INTERFACE)
+
+#Add include directories to the interface library.
+#If we use FakeIt from build-tree then we use BUILD_INTERFACE wrapper.
+#If we use FakeIt from install-tree then we use INSTALL_INTERFACE wrapper.
+target_include_directories(FakeIt INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/single_header/standalone>
+    $<INSTALL_INTERFACE:include>
+)
+
+#Add alias for the library with namespace.
+add_library(FakeIt::FakeIt ALIAS FakeIt)
+
+#Install single_header file to install directory to include subdirectory.
+#Trailing slash at "single_header/standalone/" string means that
+#the header will be installed like ${CMAKE_INSTALL_PREFIX}/include/fakeit.hpp
+#instead of ${CMAKE_INSTALL_PREFIX}/include/standalone/fakeit.hpp.
+install(DIRECTORY single_header/standalone/ DESTINATION include)
+
+#Add FakeIt target to the FakeItConfig.cmake file
+#which will be used to find FakeIt from outside.
+install(TARGETS FakeIt EXPORT FakeItConfig)
+
+#Install FakeItConfig.cmake file to the ${CMAKE_INSTALL_PREFIX}/share/FakeIt/cmake directory
+#and all correlated targets should be called from outside by the namespace.
+install(EXPORT FakeItConfig NAMESPACE FakeIt:: DESTINATION share/FakeIt/cmake)
+
+#Add version control.
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file("${PROJECT_BINARY_DIR}/FakeItConfigVersion.cmake"
+    VERSION
+        ${PROJECT_VERSION}
+    COMPATIBILITY
+        AnyNewerVersion
+)
```

Comment:
`write_basic_package_version_file` also inserts an architecture check, which we don't want as FakeIt is header-only. If we were willing to require CMake 3.14 to build the FakeIt package (we could e.g. supply a pre-built package in the releases, and users could use a lower version), the fix would be to just add `ARCH_INDEPENDENT` to the call. (documentation: https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file )

There is a hacky workaround: the check reads from `CMAKE_SIZEOF_VOID_P`. We can workaround this architecture check by setting `CMAKE_SIZEOF_VOID_P` to empty. See Catch2's use of this workaround: https://github.com/catchorg/Catch2/blob/87b745da6625b7469e61de5910978a097caf8b74/CMakeLists.txt#L140-L157

Reply:
I have added ARCH_INDEPENDENT option to write_basic_package_version_file call but it required 3.14 version of CMake. So I have changed CMake minimum required version from 3.0 to 3.14. I think it's not critical because before there was no CMake support and no projects which depend on FakeIt will be broken.